src/lib/swapper/swappers/ThorchainSwapper/utils/makeMemoWithShortenedFinalAssetAmount.ts

```diff
@@ -0,0 +1,98 @@
+import assert from 'assert'
+import { assertAndProcessMemo } from 'lib/utils/thorchain/memo'
+
+import { MEMO_PART_DELIMITER } from './constants'
+
+// The THORChain aggregators expects two numbers at the end which are used at exponents in the contract
+// see _parseAmountOutMin in https://dashboard.tenderly.co/tx/mainnet/0xe16c61c114c5815555ec4342af390a1695d1cce431a8ff479bcc7d63f3d0b46a?trace=0.2.10
+// 7759345496513 => 775934549650000000000000
+export const AGGREGATOR_EXPONENT_LENGTH = 2
+
+// THORChain uses `big.ParseFloat` with `0` as a precision, resulting in a precision of 18 without the integer part
+// https://gitlab.com/thorchain/thornode/-/blob/v1.131.0/x/thorchain/memo/memo_parser.go#L190
+export const THORCHAIN_PARSER_MAXIMUM_PRECISION = 18
```

Comment:
Looking at this more, I don't think this is an actual constraint. It should support a uint256 for increased precision if we have room for it in the memo.

```
func UintOverflow(i *big.Int) error {
	if i.Sign() < 0 {
		return errors.New("non-positive integer")
	}
	if i.BitLen() > 256 {
		return fmt.Errorf("bit length %d greater than 256", i.BitLen())
	}
	return nil
}
```

Reply:
I see your point now regarding loss of fractional value resulting in rounding issues. So it does support larger numbers, but precision is lost.